src/datachain/toolkit/split.py

```diff
@@ -56,16 +67,28 @@ def train_test_split(dc: DataChain, weights: list[float]) -> list[DataChain]:
     if any(weight < 0 for weight in weights):
         raise ValueError("Weights should be non-negative")
 
-    weights_normalized = [weight / sum(weights) for weight in weights]
+    def scale(multiplier: float) -> int:
+        if multiplier < 0:
+            multiplier = 0.0
+        if multiplier > 1:
+            multiplier = 1.0
```

Comment:
It does not in any case I can imagine, but I've added it just in case, otherwise SQLite will throw an exception (let's say if we pass `9223372036854775808` (`2^63`) instead of `9223372036854775807` (`2^63-1`), — this is, by the way, why we need this function, because in Python `int(9223372036854775807 * 1.0) == 9223372036854775808` (same for `round`) and SQLite is going crazy about this.